include/TPP/Dialect/Tpp/TppOps.td

```diff
@@ -84,21 +100,14 @@ def Tpp_IdentityOp : Tpp_Op<"identity"> {
     ```
   }];
 
-  let arguments = (ins TppOperand:$input, TppOperand:$output);
-
-  let assemblyFormat = [{
-      `ins` `(` $input `:` type($input) `)`
-      `out` `(` $output `:` type($output) `)` attr-dict
-  }];
-
   let hasVerifier = 1;
 }
 
 //===----------------------------------------------------------------------===//
 // ReluOp
 //===----------------------------------------------------------------------===//
 
-def Tpp_ReluOp : Tpp_Op<"relu"> {
+def Tpp_ReluOp : Tpp_UnaryOp<"relu", [SameTypeOperands]> {
   let summary = "Applies a Rectified Linear Unit function in place.";
```

Comment:
I think some verifier should check that input and output are the same destination for unary ops.
As an example, currently `tpp.relu` happily accepts two different arguments and further lowering just implicitly ignores the input.

Reply:
Slightly modified case from `tpp-to-loops` that will pass:
```// CHECK: func.func @relu_to_loops(
func.func @relu_to_loops(%arg0: memref<3x3xf32>, %arg1: memref<3x3xf32>) {
  // CHECK-DAG: %[[ub:.*]] = arith.constant 3 : index
  // CHECK-DAG: %[[lb:.*]] = arith.constant 0 : index
  // CHECK-DAG: %[[step:.*]] = arith.constant 1 : index
  // CHECK-DAG: %[[relu:.*]] = arith.constant 0.000000e+00 : f32
  // CHECK: scf.for %[[i:.*]] = %[[lb]] to %[[ub]] step %[[step]] {
  // CHECK:   scf.for %[[j:.*]] = %[[lb]] to %[[ub]] step %[[step]] {
  // CHECK:     %[[load:.*]] = memref.load %[[ARG0]][%[[i]], %[[j]]] : memref<3x3xf32>
  // CHECK:     %[[max:.*]] = arith.maxf %[[load]], %[[relu]] : f32
  // CHECK:     memref.store %[[max]], %[[ARG0]][%[[i]], %[[j]]] : memref<3x3xf32>
  tpp.relu ins(%arg1: memref<3x3xf32>) out(%arg0: memref<3x3xf32>)
  return
}
```

Reply:
ops that's wrong the load should be from %arg1 and the store to %arg0. thx

Reply:
So, it is no longer in place only op, right? If that's the case, I think the TableGen docs need slight updates too.

Reply:
yes, is not longer in-place but can be if you pass the same subview/memref as input and output.

Reply:
will update the .td doc.

Reply:
The point is that you can still have in-place beh. if you pass ins and out the same memref. But the new spec is 1 input and 1 output.